runtime/lua/vim/lsp/handlers.lua

```diff
@@ -81,22 +81,38 @@ M['window/workDoneProgress/create'] = function(_, result, ctx)
 end
 
 --see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#window_showMessageRequest
+---@param result lsp.ShowMessageRequestParams
 M['window/showMessageRequest'] = function(_, result)
-  local actions = result.actions
-  print(result.message)
-  local option_strings = { result.message, '\nRequest Actions:' }
-  for i, action in ipairs(actions) do
-    local title = action.title:gsub('\r\n', '\\r\\n')
-    title = title:gsub('\n', '\\n')
-    table.insert(option_strings, string.format('%d. %s', i, title))
-  end
-
-  -- window/showMessageRequest can return either MessageActionItem[] or null.
-  local choice = vim.fn.inputlist(option_strings)
-  if choice < 1 or choice > #actions then
-    return vim.NIL
+  local actions = result.actions or {}
+  local co = coroutine.running()
+  if co then
```

Comment:
Note this doesn't work in Lua 5.2 which never returns nil.

Reply:
Wouldn't the code still work, as it would return the main coroutine, and then yield/resume on that? And I thought we're only targeting luajit and lua 5.1

I also wasn't sure if the alternative path should even be kept. The neovim code should always hit the new path, but keeping the old code would keep compatibility in case someone decided to call the handler directly from their custom code.

Reply:
You can't yield from the main thread.

Reply:
Added a check for the main coroutine